.github/CONTRIBUTING.md

```diff
@@ -40,23 +63,56 @@ We expect code contributions for:
 > code!  Legacy takes precedence, and remember, we read code a lot more
 > than write it, so legibility is important.
 
-As a final note, lines are allowed to be longer than 72 characters these
-days.  There is no enforced maximum, but the team usually keep it around
-100 characters for both C and Python.
+The ChangeLog deserves a separate mention:
+
+ - Releases are listed in reverse chronological order order, so the
+   latest/next release is at the beginning of the file
+ - Only *user-facing bugs and features* are detailed, so code refactor,
+   new tests, etc. are not listed.
+ - Add your changes/features to the Changes section
+ - Add your Fix line in the Fixes section, in numeric order 
+ - Changes and fixes without an issue number are listed after all
+   numbered ones
+ - YANG model changes are documented in their respective model, for
+   standard models, e.g., for `ietf-interfaces.yang`, the corresponding
+   `infix-interfaces.yang` detail augments/deviations as revisions.
+
+A final note, lines of code are allowed to be longer than 72 characters
+these days, unless you live by PEP-8 (see above).  There is no enforced
+maximum, but the team usually keep it around 100 characters for both C
+and Python.
+
+
+:test_tube: Testing Changes
+---------------------------
+
+Please test your changes, no matter how trivial or obviously correct
+they may seem.  Nobody is infallible, making mistakes is only human.
```

Comment:
```suggestion
they may seem.  Nobody is infallible, making mistakes is only human.
It is also the best insurance policy for your feature, since it
ensures that your use-case will remain functional as the source base
evolves.
```